src/main/java/org/intellij/sdk/codesync/clients/JSONResponse.java

```diff
@@ -78,10 +78,13 @@ public void raiseForStatus () throws StatusCodeError {
                 if (errorObject != null && errorObject.containsKey("message")) {
                     errorMessage = (String) errorObject.get("message");
                 }
-                if (errorObject != null && errorObject.containsKey("error_code")) {
-                    errorCode = ((Number) errorObject.get("error_code"));
-                    if (errorCode != null){
-                        customErrorCode = errorCode.intValue();
+                // Extract error_code only if a 402 is received
+                if(this.statusCode == 402){
```

Comment:
Better to define a constant like we did above on line 72 as ```Constants.ErrorCodes.ACCOUNT_DEACTIVATED```

Reply:
We seem to have a constant for status_code 402, but with a different name.

![image](https://github.com/codesyncapp/codesync-intellij/assets/15974895/53a39579-7141-4c9b-87fc-ccec87069458)

I was thinking we create a new class, "FreePlanErrors" or "PricingErrors", and name this `PAYMENT_REQUIRED`, as we have in our server.

Thoughts?

Reply:
We can do that. or we can also rename the existing one.
`DIFFS_LIMIT_REACHED` is not being raised by server anymore, I think we can remove it from the Plugin.

Reply:
I'll rename the existing one then, and rename this only occurrence of `DIFFS_LIMIT_REACHED` to `PAYMENT_REQUIRED`

![image](https://github.com/codesyncapp/codesync-intellij/assets/15974895/4ee78606-fe34-4567-a91d-d4cb995a8d14)